kernel/src/transaction/create_table.rs

```diff
@@ -0,0 +1,991 @@
+    /// * `committer` - The committer to use for the transaction
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - The table path is invalid
+    /// - A table already exists at the given path
+    /// - The schema is empty
+    /// - Clustering columns don't exist in schema or exceed the limit (4)
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use delta_kernel::table_manager::TableManager;
+    /// # use delta_kernel::schema::{StructType, DataType, StructField};
+    /// # use delta_kernel::committer::FileSystemCommitter;
+    /// # use std::sync::Arc;
+    /// # use delta_kernel::Engine;
+    /// # fn example(engine: &dyn Engine) -> delta_kernel::DeltaResult<()> {
+    /// # let schema = Arc::new(StructType::try_new(vec![StructField::new("id", DataType::INTEGER, false)]).unwrap());
+    /// let txn = TableManager::create_table("/path/to/table", schema, "MyApp/1.0")
+    ///     .build(engine, Box::new(FileSystemCommitter::new()))?;
+    ///
+    /// // Commit the transaction to create the table
+    /// let result = txn.commit(engine)?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn build(
+        self,
+        engine: &dyn Engine,
+        committer: Box<dyn Committer>,
+    ) -> DeltaResult<Transaction> {
+        // Validate path
+        let table_url = try_parse_uri(&self.path)?;
+
+        // Validate schema is non-empty
+        if self.schema.fields().len() == 0 {
+            return Err(Error::generic("Schema cannot be empty"));
+        }
+
+        // Validate and strip protocol version properties (before engine access for fail-fast)
+        let table_properties = Self::validate_and_strip_protocol_properties(self.table_properties)?;
+
+        // Ensure no table exists at the path
+        Self::ensure_table_does_not_exist(&table_url, &self.path, engine)?;
+
+        // Extract feature overrides and build reader/writer feature lists
+        let mut extracted = Self::extract_table_features(table_properties)?;
+
+        // Process column mapping FIRST (before data layout, as clustering needs physical names)
+        let (schema_with_cm, column_mapping_mode) =
+            Self::apply_column_mapping(&self.schema, &mut extracted)?;
+
+        // Process data layout (partitioning or clustering) using the updated schema
+        let layout =
+            Self::process_data_layout(&self.data_layout, &schema_with_cm, column_mapping_mode)?;
+
+        // Add any features required by the data layout
+        for feature in layout.additional_writer_features {
+            if !extracted.writer_features.contains(&feature) {
+                extracted.writer_features.push(feature);
+            }
+        }
+
+        // Create Protocol action with table features support
+        let protocol = Protocol::try_new(
+            TABLE_FEATURES_MIN_READER_VERSION,
+            TABLE_FEATURES_MIN_WRITER_VERSION,
+            Some(extracted.reader_features),
```

Comment:
I think we need to leave a TODO: this does not yet enable required (downstream; dependent) properties, right? like how rowTracking required domainMetadata?

aside: can we detect this and throw?

cc @DrakeLin